rpy/rinterface_lib/sexp.py

```diff
@@ -236,7 +237,10 @@ def nchar(self, what: NCHAR_TYPE = NCHAR_TYPE.Bytes) -> int:
 # will have API-defined code compile for efficiency).
 def _populate_r_vector(iterable, r_vector, set_elt, cast_value):
     for i, v in enumerate(iterable):
-        set_elt(r_vector, i, cast_value(v))
+        if type(v) is float and np.isnan(v):
+            set_elt(r_vector, i, openrlib.rlib.R_NaInt)
```

Comment:
I probably did something wrong, but this does solve my issue :).

I added a test that converts a nullable pandas array and from there modified the failing code. Trough this statement [1], the conversion of the pandas dataframe is moved over to the `numpy` parsing:

[1] https://github.com/rpy2/rpy2/blob/master/rpy/robjects/pandas2ri.py#L141

Is that not supposed to happen?

I added `type(v) is float` since `np.isnan(v)` would sometimes fail if `v` is not a float. The type of the `NaN` value in the int array is actually set to float by pandas.

doing a condition in this loop is probably slow, but I wasn't sure how to do this otherwise.

Reply:
What I mean is that if `iterable` only contains `float` items, the ones of them that are `nan` shouldn't probably be cast into R's NA for integers.

Regarding the test in the loop, there is iittle that can be done outside of moving the loop to C (which is planned, as it was the case before the port to `cffi`).

Reply:
I think this is better now. I've now registered a `nullable_int` type to the `IntSexpVector` class that will check for NaN floats in the array.

Reply:
does this answer your question now also? Nullable int arrays in pandas contain ints but a float NaN when data is missing. So I'm quite sure they should just become R int arrays.

Reply:
Thanks. An issue with the update is that `numpy` is an optional dependency for `rpy2`: the `import numpy` statements sprinkled throughout the PR are breaking.

Reply:
what do you propose? i can make the import optional, but the `np.isnan` check is required.

Reply:
I believe that testing for `NaN` does not require `numpy`. Wouldn't `math.isnan` work?

Reply:
good point, yes it works. changed.